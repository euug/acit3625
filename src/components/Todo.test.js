import React from 'react';
import renderer from 'react-test-renderer';
import Todo from './Todo';
import { shallow } from 'enzyme';
import "../setupTests";


describe('Render Tests', () => {
    test('Snapshot Test', () => {
        const component = renderer.create(<Todo />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('Renders editingTemplate if isEditing State is True', () => {
        const component = shallow(<Todo/>);
      
        component.find('#edit-button').simulate('click');
        component.update();
        
        expect(component.find('#cancel-button').text()).toEqual('Cancelrenaming ')
    });
})
