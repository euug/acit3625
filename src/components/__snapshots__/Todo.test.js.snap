// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Component Tests Renders Correctly 1`] = `
<li
  className="todo"
>
  <div
    className="stack-small"
  >
    <div
      className="c-cb"
    >
      <input
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="todo-label"
      />
    </div>
    <div
      className="btn-group"
    >
      <button
        className="btn"
        id="edit-button"
        onClick={[Function]}
        type="button"
      >
        Edit 
        <span
          className="visually-hidden"
        />
      </button>
      <button
        className="btn btn__danger"
        onClick={[Function]}
        type="button"
      >
        Delete 
        <span
          className="visually-hidden"
        />
      </button>
    </div>
  </div>
</li>
`;
